code/modules/clothing/clothing.dm

```diff
@@ -103,6 +122,124 @@
 	else
 		icon = initial(icon)
 
+//material related procs
+
+/obj/item/clothing/get_material()
+	return material
+
+/obj/item/clothing/proc/set_material(var/new_material)
+	material = get_material_by_name(new_material)
+	if(!material)
+		qdel(src)
+	else
+		name = "[material.display_name] [initial(name)]"
+		health = round(material.integrity/10)
+		if(applies_material_color)
+			color = material.icon_colour
+		if(material.products_need_process())

```

Comment:
I'm not seeing an actual process() override implemented in this PR? What does this bit even doooo.

Reply:
It is used by some materials, such as uranium.